back/boxtribute_server/graph_ql/resolvers.py

```diff
@@ -532,6 +698,39 @@ def resolve_base_locations(base_obj, _):
     return Location.select().where(Location.base == base_obj.id)
 
 
+@base.field("distributionSpots")
+@query.field("distributionSpots")
+def resolve_distributions_spots(base_obj, _):
+    authorize(permission="location:read")
+    return (
+        Location.select()
+        .where(Location.type == LocationType.DistributionSpot)
+        .where(base_filter_condition("location:read"))
+    )
+
+
+@query.field("distributionSpot")
+def resolve_distributions_spot(obj, _, id):
+    distribution_spot = obj.location if id is None else Location.get_by_id(id)
```

Comment:
the `id` argument will always be given, so you can shorten this to `distribution_spot = Location.get_by_id(id)` and drop the `obj` argument. Same for the distroEvent query below